api/features/feature_segments/serializers.py

```diff
@@ -63,21 +64,65 @@ def get_is_feature_specific(self, instance: FeatureSegment) -> bool:
         return instance.segment.feature is not None
 
 
-class FeatureSegmentChangePrioritiesSerializer(serializers.Serializer):
+class FeatureSegmentChangePrioritiesListSerializer(serializers.ListSerializer):
+    def validate(self, attrs):
+        validated_attrs = super().validate(attrs)
+        feature_segments = list(
+            FeatureSegment.objects.filter(
+                id__in=[item["id"] for item in validated_attrs]
+            )
+        )
+
+        # TODO: what if length == 0?
+        if not len(feature_segments) == len(attrs):
+            raise serializers.ValidationError(
+                "Some of the provided ids were not found."
+            )
+
+        environments = set()
+        features = set()
+
+        for feature_segment in feature_segments:
+            environments.add(feature_segment.environment)
+            features.add(feature_segment.feature)
+
+        if not len(environments) == len(features) == 1:
+            raise serializers.ValidationError(
+                "All feature segments must belong to the same feature & environment."
+            )
+
+        return validated_attrs
+
+    def create(self, validated_data):
+        id_priority_pairs = FeatureSegment.to_id_priority_tuple_pairs(validated_data)
+        (
+            changed,
+            previous_id_priority_pairs,
+            feature_segments,
+        ) = FeatureSegment.update_priorities(id_priority_pairs)
+
+        if changed:
+            create_segment_priorities_changed_audit_log.delay(
```

Comment:
I don't love that this is happening in the serializer, however, I put it here for 2 reasons: 

1. I need access to the user / master api key
2. putting it in `models` caused a circular import

I thought about raising it another level to the view, however, that would mean either: (a) calling the `update_priorities` method in the view and just using the serializer for validation; or (b) returning more information from this `create` method. 

Interested to know what your thoughts are on this @gagantrivedi .

Reply:
I think Ideally it should be in `models`, but out of the two I am leaning slightly towards doing it in serializer (because it does not directly affect the data being returned, in the view layer sense)

Reply:
Yeah, the issue with putting it in models was the circular import. I _could_ use a local import but that makes testing pretty tough. I'll have another look at it this morning but I'm not sure if I can get it working. The other option I considered was creating a custom signal for when priorities are changed?

Reply:
Hmm yeah, tough choice. But I think even with that, the logic for audit logs will still be at multiple places. Considering that, I don't think it's too right now

Reply:
Yeah, I have tried to chase down the circular imports and tidy it up but I'm not sure if it's better or worse? 

https://github.com/Flagsmith/flagsmith/pull/1917/commits/3962504bc8de76fd1b2643e09443e1627343f0a3

Reply:
Ah yeah, but I think prefer this because it keeps the audit log logic in models at least, no?

Reply:
Yep, I think on the whole it is a little neater.